internal/test/framework/framework.go

```diff
@@ -174,6 +189,111 @@ func (f *TestFramework) getOpenShiftConfigClient() error {
+// oc get clusterversion, as OpenShift CI doesn't have the actual version attached to its clusters, instead replacing it
+// with 0.0.1 and information about the release creation date
+func (f *TestFramework) getClusterVersion() error {
+	versionInfo, err := f.OSConfigClient.Discovery().ServerVersion()
+	if err != nil {
+		return err
+	}
+
+	// Convert kubernetes version to major.minor format. v1.17.0 -> 1.17
+	k8sVersion := strings.TrimLeft(versionInfo.GitVersion, "v")
+	k8sSemver := strings.Split(k8sVersion, ".")
+	k8sMinorVersion, err := strconv.Atoi(k8sSemver[1])
+	if err != nil {
+		return fmt.Errorf("could not conver k8s minor version %s to int", err)
+	}
+
+	// Map kubernetes version to OpenShift version
+	openshiftVersion, err := k8sVersionToOpenShiftVersion(k8sMinorVersion)
+	if err != nil {
+		return fmt.Errorf("could not map kubernetes version to an OpenShift version: %s", err)
+	}
+
+	f.ClusterVersion = openshiftVersion
+	return nil
+}
+
+// getLatestGithubRelease gets the latest github release for the wmcb repo. This release is specific to the cluster
+// version
+func (f *TestFramework) getLatestGithubRelease() error {
+	client := github.NewClient(nil)
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*20)
+	defer cancel()
+	releases, _, err := client.Repositories.ListReleases(ctx, "openshift", "windows-machine-config-operator",
+		&github.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	for _, release := range releases {
+		if strings.Contains(release.GetName(), f.ClusterVersion) {
+			f.latestRelease = release
+			return nil
+		}
+	}
+	return fmt.Errorf("could not fetch latest release")
+}
+
+// GetLatestReleaseArtifactURL returns the URL of the releases artifact matching the given name
+func (f *TestFramework) GetReleaseArtifactURL(artifactName string) (string, error) {
+	for _, asset := range f.latestRelease.Assets {
+		if asset.GetName() == artifactName {
+			return asset.GetBrowserDownloadURL(), nil
+		}
+	}
+	return "", fmt.Errorf("no artifact with name %s", artifactName)
+}
+
+// GetReleaseArtifactSHA returns the SHA256 of the release artifact specified, given the body of the release
+func (f *TestFramework) GetReleaseArtifactSHA(artifactName string) (string, error) {
+	// The release body looks like:
+	// f819c2df76bc89fe0bd1311eea7dae2a11c40bc26b48b85fd4718e286b0a257e  wmcb.exe
+	// 92c24c250ef81b565e2f59916f722d8fcb0a5ec1821899d590b781e3c883a7d3  wni
+	// a476f9b7e8b223f5d2efb2066ae48be514d57b66e04038308d1787a770784084  hybrid-overlay.exe
+	lines := strings.Split(f.latestRelease.GetBody(), "\r\n")
+	for _, line := range lines {
+		// Get the line that ends with the artifact name
+		if strings.HasSuffix(line, " "+artifactName) {
+			return strings.Split(line, " ")[0], nil
+		}
+	}
+	return "", fmt.Errorf("no artifact with name %s", artifactName)
+}
+
+// GetNode returns a pointer to the node object associated with the external IP provided
```

Comment:
Please ignore this function as it will get added in #140 at which point I will remove it from here.